boring.js

```diff
@@ -1,18 +1,26 @@
-module.exports = function() {
+module.exports = function(options) {
+  options = options || {};
   const args = process.argv.slice(2);
   const result = {
     _: []
   };
+  let optionsEnded = false;
   for (let i = 0; (i < args.length); i++) {
-    let matches = args[i].match(/^--([^\s\=]+)=(.*)$/);
-    if (matches) {
-      result[matches[1]] = matches[2];
+    if (options.end && (args[i] === options.end)) {
```

Comment:
Is it better for this to be an option option, or to define that double dashes are the ender indicator? If it's common elsewhere it seems a reasonable ender. Or default to that with `true` and allow an alternate if the value is a string.